src/ray/pubsub/subscriber.cc

```diff
@@ -111,23 +111,23 @@ void SubscriberChannel<KeyIdType>::HandlePublisherFailure(
   }
   auto &subscription_callback_map = subscription_it->second.subscription_callback_map;
 
-  std::vector<KeyIdType> key_ids_to_unsubscribe;
+  std::vector<std::string> key_ids_to_unsubscribe;
   for (const auto &key_id_it : subscription_callback_map) {
     const auto &key_id = key_id_it.first;
-    key_ids_to_unsubscribe.push_back(key_id);
+    key_ids_to_unsubscribe.push_back(key_id.Binary());
 
     auto maybe_failure_callback = GetFailureCallback(publisher_address, key_id);
     if (maybe_failure_callback.has_value()) {
       const auto &failure_callback = maybe_failure_callback.value();
-      failure_callback();
+      failure_callback(key_id.Binary());
```

Comment:
how about: `failure_callback(key_ids_to_unsubscribe.back())`

Reply:
Sure. Is it to reduce the copy overhead from key_id.Binary?

Reply:
Yes, probably no improvement in performance, but it also doesn't hurt.